lambdas/models/nhs_document_reference.py

```diff
@@ -44,16 +45,10 @@ def __init__(
         self.last_updated = int(date_now.timestamp())
 
     def set_file_location(self):
-        file_location = f"s3://{self.s3_bucket_name}"
-
         if self.sub_folder != "":
-            file_location += f"/{self.sub_folder}"
-
-        if self.doc_type != "":
-            file_location += f"/{self.doc_type}"
-
-        file_location += f"/{self.s3_file_key}"
-        return file_location
+            return f"s3://{self.s3_bucket_name}/{self.sub_folder}/{self.doc_type}/{self.nhs_number}/{self.id}"
```

Comment:
The previous code was more configurable to allow for more path forms such as 

bucket/sub/doc...
bucket/sub....
bucket/doc...

This change now limits that and turns future changes into a state configuration rather than a route builder. I would revert this. 

I also discourage having your return data set on the return line without a variable before hand.
It's trickier to debug from a break point perspective and if you fall into a pattern of returning function response, code debugging becomes difficult to say the least. The minor memory consideration does is less useful than the code maintainability.

e.g.

rhyme = "Humpty Dumpty sat on the wall" 
return rhyme

vs 

return "Humpty Dumpty sat on the wall"